### PR TITLE
Implement Index.drop

### PR DESCRIPTION
**Pull request by RainFung**
Implement Index.drop by using spark isin API

### CODE REVIEW

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -580,6 +580,33 @@ def unique(self, level=None):
...
+        >>> index = ks.Index([1, 2, 3])
+        >>> index
+        Int64Index([1, 2, 3], dtype='int64')
+
+        >>> index.drop([1])
+        Int64Index([2, 3], dtype='int64')
+        """
+        if not isinstance(labels, (tuple, list)):
+            labels = [labels]
+        sdf = self._internal.sdf[~self._internal.index_scols[0].isin(labels)]
```
Comment (itholic) [inline databricks/koalas/indexes.py]: i think maybe you can use below code to cover Index/MultiIndex cases :)

```python
sdf = self._internal.sdf
conds = []
for index_scol, label in zip(self._internal.index_scols, labels):
    conds.append(index_scol == label)

cond = ~reduce(lambda x, y: x & y, conds)
sdf.where(cond)
```

Comment (HyukjinKwon) [reply databricks/koalas/indexes.py]: Yeah, @RainFung. I think this comment is correct. Can you update please accordingly?

Comment (HyukjinKwon) [reply databricks/koalas/indexes.py]: ping @RainFung

File: databricks/koalas/tests/test_indexes.py [hunk 2]
```diff
@@ -391,3 +408,4 @@ def test_multiindex_rename(self):
         pidx = pidx.rename(['my', 'name', 'is'])
         kidx = kidx.rename(['my', 'name', 'is'])
         self.assert_eq(pidx, kidx)
+
```
Comment (HyukjinKwon) [inline databricks/koalas/tests/test_indexes.py]: Can we remove this line?

**Pull request merged**